diff --git a/libcfs/include/libcfs/linux/linux-time.h b/libcfs/include/libcfs/linux/linux-time.h
--- a/libcfs/include/libcfs/linux/linux-time.h
+++ b/libcfs/include/libcfs/linux/linux-time.h
@@ -235,5 +235,9 @@
 #ifdef HAVE_TIMER_SETUP
+#ifndef timer_container_of
+#define timer_container_of(var, callback_timer, timer_fieldname) \
+	container_of(callback_timer, typeof(*var), timer_fieldname)
+#endif
 #define cfs_timer_cb_arg_t struct timer_list *
 #define cfs_from_timer(var, callback_timer, timer_fieldname) \
-	from_timer(var, callback_timer, timer_fieldname)
+	timer_container_of(var, callback_timer, timer_fieldname)
 #define cfs_timer_setup(timer, callback, data, flags) \
